### ISSUE TITLE
No error on undefined variables

### ISSUE THREAD

**Issue by Mottie** [stylelint/stylelint#2837]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Undefined variables should report an error. Enter the following into the online demo and it reports "No warnings"

```css
/* should show an error */
a { color: var(--undef); }

/* this variable has a fallback, but should also show an error if the config is set */
a { color: var(--undef, red); }

/* this variable has nested fallbacks, but should also show an error if the config is set */
a { color: var(--undef, var(--undef2, red)); }

```

When the browser evaluates these variables, it sets the color to `rgba(0, 0, 0, 0);` for the first entry, the other two fallback to red.

> Which rule, if any, is this issue related to?

A new rule would be required.

> What CSS is needed to reproduce this issue?

See above

> What stylelint configuration is needed to reproduce this issue?

Using the [online demo](https://stylelint.io/demo/) default settings

> Which version of stylelint are you using?

`8.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Online demo

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Nope

> What did you expect to happen?

A warning or error should notify of an undefined variable.

> What actually happened (e.g. what warnings or errors you are getting)?

Nada

**Comment by jeddy3** [comment]
@Mottie Thanks for the suggestion. Labeling as `plugin`. Feel free to [create](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md) `no-unknown-custom-properties` as a plugin). See https://github.com/stylelint/stylelint/issues/2220#issuecomment-284677340.

**Comment by Mottie** [comment]
It might be nice to have a plugin repository where we can transfer these rule proposals. When they're closed here, people don't usually go searching for rule ideas, when they get the urge to work on something 😉.

**Comment by jeddy3** [comment]
> It might be nice to have a plugin repository where we can transfer these rule proposals.

It's a good suggestion and I see the value in it, but I don't think it adds enough to warrant the ongoing effort as it's for a small audience. There's already so much to manage, maintain and juggle that we're consciously streamlining and reducing things-we-have-to-maintain.  We had a wishlist on the wiki but it wasn't updated.
